File: python/268.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        for i in range(n):
            while nums[i] != i:
                j = nums[i]
                if j == n:
                    break
                nums[i], nums[j] = nums[j], nums[i]
        print(nums)
        for i in range(n):
            if i != nums[i]:
                return i
        return n

sol = Solution()
print(sol.missingNumber([3, 2, 0]))

File: cpp/268.cpp
#include <vector>
using namespace std;

class Solution {
public:
  int missingNumber(vector<int> &nums) {
    int n = nums.size();
    for (int i = 0; i < n; i++) {
      while (nums[i] != i) {
        int j = nums[i];
        if (j == n)
          break;
        int t = nums[j];
        nums[j] = nums[i];
        nums[i] = t;
      }
    }
    for (int i = 0; i < n; i++)
      if (nums[i] != i)
        return i;
    return n;
  }
};
